dom/events/scrolling/wheel-event-transactions-target-removal.html

```diff
@@ -0,0 +1,92 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+<meta name="viewport" content="width=device-width,initial-scale=1,minimum-scale=1">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script src="scroll_support.js"></script>
+<style>
+body {
+  margin: 0;
+  padding: 0;
+  height: 200vh;
+}
+
+.spacer {
+  width: 100%;
+  height: 25px;
+  padding: 0;
+  margin: 0;
+}
+
+</style>
+<head>
+<body>
+<div id="initialTarget" class="spacer" style="background: red"></div>
+<div id="firstRootSpacer" class="spacer" style="background: green"></div>
+<div id="secondRootSpacer" class="spacer" style="background: blue"></div>
+</body>
+
+<script>
+
+promise_test(async (t) => {
+  await new Promise(resolve => addEventListener("load", resolve, {once: true}));
+
+  await waitForCompositorReady();
+
+  await waitForCompositorCommit();
+
+
+  let initialTarget = null;
+  let wheelEventTargets = [];
+  let removedInitialTarget = false;
+
+  // Remove the initial element the wheel event is targetted at once we fire the
+  // first wheel event, then log the subsequent wheel event targets.
+  function removeInitialElement(e) {
+    wheelEventTargets.push(e.target);
+    if (!removedInitialTarget) {
+      initialTarget = e.target;
+      e.target.remove();
+      removedInitialTarget = true;
+    }
+  }
+  window.addEventListener("wheel", removeInitialElement, {passive: true});
+
+  await waitForCompositorCommit();
+
+  await new test_driver.Actions()
+    .addWheel("wheel1")
+    .scroll(40, 2, 0, 30, {origin: "viewport"})
+    .pause(1)
+    .scroll(40, 2, 0, 30, {origin: "viewport"})
+    .send();
+
+  // TODO(dlrobertson): Use the scrollend event here to wait for the
+  // wheel scroll to finish instead of waitForAnimationEnd().
+  await waitForAnimationEnd(() => { return document.scrollingElement.scrollTop; });
+  await waitForCompositorCommit();
+
+  initialTarget.addEventListener("wheel", () => {
+    assert_true(false, "wheel event should never be fired after the target is removed")
```

Comment:
I don't understand this test. Looks like that you add this event listener after all events are fired. Don't you need to do this in the first `wheel` event listener, `removeInitialElement`?

Reply:
Good catch! Updated to set this in the first wheel event listener.